linera-views/src/hashable_wrapper.rs

```diff
@@ -72,23 +72,22 @@ where
 
     fn flush(&mut self, batch: &mut Batch) -> Result<bool, ViewError> {
         let delete_view = self.inner.flush(batch)?;
+        let hash = self.hash.get_mut();
         if delete_view {
             let mut key_prefix = self.inner.context().base_key();
             key_prefix.pop();
             batch.delete_key_prefix(key_prefix);
             self.stored_hash = None;
-        } else {
-            let hash = *self.hash.get_mut();
-            if self.stored_hash != hash {
-                let mut key = self.inner.context().base_key();
-                let tag = key.last_mut().unwrap();
-                *tag = KeyTag::Hash as u8;
-                match hash {
-                    None => batch.delete_key(key),
-                    Some(hash) => batch.put_key_value(key, &hash)?,
-                }
-                self.stored_hash = hash;
+            *hash = None;
+        } else if self.stored_hash != *hash {
+            let mut key = self.inner.context().base_key();
+            let tag = key.last_mut().unwrap();
+            *tag = KeyTag::Hash as u8;
```

Comment:
Not from this PR but we're really missing a comment to explain why it's ok to abuse the base key of the inner object.